### PR TITLE
alertmanager: handle resolved alerts, nodata, and execution errors + cleanups

### PR DESCRIPTION
**Pull request by iksaif**
- Cleanup notifiers
- Resolved alerts will be sent to the AM the first time they move
  from Alerting to OK
- Errors are forwarded
- NoData is not ignored anymore
- The summary annotation is set (with the alertname)
- The image annotation is set (when available)

### CODE REVIEW

File: pkg/services/alerting/notifiers/base.go [hunk 1]
```diff
@@ -27,15 +27,21 @@ func NewNotifierBase(id int64, isDefault bool, name, notifierType string, model
...
 		return false
 	}
+	// Do not notify when we become OK for the first time.
 	if (context.PrevAlertState == m.AlertStatePending) && (context.Rule.State == m.AlertStateOK) {
 		return false
 	}
 	return true
 }
 
+func (n *NotifierBase) ShouldNotify(context *alerting.EvalContext) bool {
```
Comment (bergquist) [inline pkg/services/alerting/notifiers/base.go]: :heart:

File: pkg/services/alerting/notifiers/alertmanager.go [hunk 2]
```diff
@@ -46,36 +46,82 @@ type AlertmanagerNotifier struct {
...
-		tags["alertname"] = evalContext.Rule.Name
-		alertJSON.Set("labels", tags)
+	}
+	tags["alertname"] = evalContext.Rule.Name
+	alertJSON.Set("labels", tags)
+	return alertJSON
+}
+
+func (this *AlertmanagerNotifier) Notify(evalContext *alerting.EvalContext) error {
+	this.log.Info("Creating Altermanager alert", "ruleId", evalContext.Rule.Id, "notification", this.Name)
```
Comment (bergquist) [inline pkg/services/alerting/notifiers/alertmanager.go]: `s/Creating/Sending` ?

**Review by bergquist — COMMENTED** [review]
Looks good! But I think the log statement could be confusing.

File: pkg/services/alerting/notifiers/alertmanager.go (hunk 2 above)
Comment (iksaif) [reply pkg/services/alerting/notifiers/alertmanager.go]: done

**Comment by iksaif** [comment]
Thanks for quick review :+1:

**Review by bergquist — APPROVED** [review]

**Comment by bergquist** [comment]
Can be merged once the build pass.

**Comment by iksaif** [comment]
Thanks :+1:

**Pull request merged**